linera-service/src/cli/validator.rs

```diff
@@ -0,0 +1,1260 @@
+        #[arg(long, short = 'y')]
+        yes: bool,
+        /// Skip online connectivity checks for validators being added or modified
+        #[arg(long)]
+        skip_online_check: bool,
+    },
+
+    /// List all validators in the committee.
+    ///
+    /// Displays the current validator set with their network addresses, voting weights,
+    /// and connection status. Optionally filter by minimum voting weight.
+    List {
+        /// Chain ID to query (defaults to default chain)
+        #[arg(long)]
+        chain_id: Option<ChainId>,
+        /// Only show validators with at least this many votes
+        #[arg(long)]
+        min_votes: Option<u64>,
+    },
+
+    /// Query a single validator's state and connectivity.
+    ///
+    /// Connects to a validator at the specified network address and queries its
+    /// view of the blockchain state, including block height and committee information.
+    Query {
+        /// Network address of the validator (e.g., grpcs://host:port)
+        address: String,
+        /// Chain ID to query about (defaults to default chain)
+        #[arg(long)]
+        chain_id: Option<ChainId>,
+        /// Expected public key of the validator (for verification)
+        #[arg(long)]
+        public_key: Option<ValidatorPublicKey>,
+    },
+
+    /// Remove a validator from the committee.
+    ///
+    /// Removes the validator with the specified public key from the committee.
+    /// The validator will no longer participate in consensus.
+    Remove {
+        /// Public key of the validator to remove
+        #[arg(long)]
+        public_key: ValidatorPublicKey,
+    },
+
+    /// Synchronize chain state to a validator.
+    ///
+    /// Pushes the current chain state from local storage to a validator node,
+    /// ensuring the validator has up-to-date information about specified chains.
+    Sync {
+        /// Network address of the validator to sync (e.g., grpcs://host:port)
+        address: String,
+        /// Chain IDs to synchronize (defaults to all chains in wallet)
+        #[arg(long)]
+        chains: Vec<ChainId>,
+        /// Verify validator is online before syncing
+        #[arg(long)]
+        check_online: bool,
+    },
+}
+
+/// Parse a batch operations file or stdin.
+/// If path is None or "-", reads from stdin.
+/// Otherwise reads from the specified file path.
+fn parse_batch_file(path: Option<&str>) -> Result<ValidatorBatchFile> {
+    let contents = match path {
+        None | Some("-") => {
+            // Read from stdin
+            use std::io::Read;
+            let mut buffer = String::new();
+            std::io::stdin()
+                .read_to_string(&mut buffer)
+                .context("Failed to read from stdin")?;
+            buffer
+        }
+        Some(path_str) => {
+            // Read from file
+            std::fs::read_to_string(path_str)
+                .with_context(|| format!("Failed to read batch file: {}", path_str))?
+        }
+    };
```

Comment:
[clio](https://crates.io/crates/clio) will do this in a generic way (failing serde parsing if the file/stream is not readable).